code/modules/mob/living/carbon/alien/adult/queen.dm

```diff
@@ -181,3 +184,28 @@
 	if(!silent)
 		to_chat(user, span_noticealien("You discard [src]."))
 	return ..()
+
+/mob/living/carbon/alien/adult/royal/queen/attackby(obj/item/O, mob/user, params)
+	. = ..()
+	if(.)
+		return
+
+	if(istype(O, /obj/item/clothing/under/costume/maid) || istype(O, /obj/item/clothing/under/rank/civilian/janitor/maid))
```

Comment:
you should either have a list of possible uniforms and check against that or have a switch or something if you plan to include more than 1 uniform

Reply:
maybe have a list with each type associated with a costume string like 
list(/obj/item/clothing/under/costume/maid = "maid",  type 2 = "costume 2")
and it sets costume = "whatever string it was associated with" then call dressup() and so on